akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingSpec.scala

```diff
@@ -39,13 +39,14 @@ object ClusterShardingSpec {
   case object Stop
   final case class CounterChanged(delta: Int)
 
-  class Counter extends PersistentActor {
+  def counterPropsFactory: String ⇒ Props = id ⇒ Props(new Counter(id))
```

Comment:
is this showing up in documentation? I think the idiom should be to define it in the companion object, as we recommend for ordinary Props also. And I don't think there is a need to define the function there.
```
object Counter {
  val ShardingTypeName = "Counter"
  def props(id: String): Props = Props(new Counter(id))
}
```

and when registering it:
```
ClusterSharding(system).start(Counter.ShardingTypeName, Counter.props, ...
```
or in docs perhaps better to spell it out:
```
ClusterSharding(system).start(Counter.ShardingTypeName, entityId => Counter.props(entityId), ...
```